packages/autocomplete-core/src/stateReducer.ts

```diff
@@ -139,14 +141,14 @@ export const stateReducer: Reducer = (action, state, props) => {
     case 'mousemove': {
       return {
         ...state,
-        highlightedIndex: action.value,
+        highlightedIndex: action.payload,
       };
     }
 
     case 'mouseleave': {
       return {
         ...state,
-        highlightedIndex: props.defaultHighlightedIndex,
+        highlightedIndex: action.props.defaultHighlightedIndex,
```

Comment:
eg. this could maybe just be the payload?

Reply:
In my mind it's still not totally clear what should be passed in our payload when thinking of it as an API. Can you explain the reasoning behind your suggestion?

Reply:
I don't really use the reducer pattern so often, but saw action.type as the identifier, and then in the cases itself, I usually only saw payload being used, so when writing the dispatch, you don't have to think whether to use payload or props or something else. 

Props of course doesn't make sense to think about, since those are the "global arguments", which makes sense as its own thing indeed

Reply:
Yep, merging as is then!